homeassistant/core.py

```diff
@@ -180,6 +180,28 @@ class HassJobType(enum.Enum):
     Executor = 3
 
 
+class Cancel:
```

Comment:
The alternative to this is just not logging an issue when unsub is called multiple times?

Reply:
The reason I didn't do this initially because it's an indicator that other code might also not be cleaned up / running twice. Now I'm open for debate if that is worth the guards in certain places.

Reply:
Also with an object, you can still write something like that. It just allows us to handle the handler with more data and a bit more intelligence as old C callback style

Reply:
> It's not duplicated in that many places I think? Most places we just do things like `self.async_on_remove(listen(…))`.

Right, that's the pattern I'm thinking about. "Everywhere" was obviously an exaggeration 😄